### PR TITLE
[Wsign-compare] warning resolutions, by directory, 6

### PR DESCRIPTION
**Pull request by tg-at-google**
Resolution affecting directories:
`tensorflow/compiler/xla`,
`tensorflow/compiler/xla/service`

@mihaimaruseac

### CODE REVIEW

**Comment by mihaimaruseac** [comment]
`tensorflow/compiler/xla` is also covered y #41752.

File: tensorflow/compiler/xla/service/llvm_ir/llvm_loop.cc [hunk 1]
```diff
@@ -276,8 +276,9 @@ std::vector<llvm::Value*> ForLoopNest::EmitOperandArrayLoopNest(
       AddLoopsForShapeOnDimensions(shape, dimensions, name_suffix);
   // Verify every dimension except the 'dimension_to_skip' dimension was set in
   // the index.
+  const int64 dimension_to_skip_int = dimension_to_skip;
   for (size_t dimension = 0; dimension < multi_index.size(); ++dimension) {
-    if (dimension == dimension_to_skip) {
+    if (dimension == dimension_to_skip_int) {
```
Comment (mihaimaruseac) [inline tensorflow/compiler/xla/service/llvm_ir/llvm_loop.cc:281]: `dimension_to_skip` is already `int64`.

File: tensorflow/compiler/xla/service/memory_space_assignment.cc [hunk 2]
```diff
@@ -298,7 +298,8 @@ CostAnalysisPrefetchIntervalPicker::CostAnalysisPrefetchIntervalPicker(
     float elapsed_time = cost_analysis_.GetInstructionElapsed(
         *instruction_and_logical_time.first);
     int64 logical_time = instruction_and_logical_time.second;
-    if (logical_time >= instructions_elapsed_time.size()) {
+    const int64 instructions_elapsed_time_size = instructions_elapsed_time.size();
+    if (logical_time >= instructions_elapsed_time_size) {
```
Comment (mihaimaruseac) [inline tensorflow/compiler/xla/service/memory_space_assignment.cc:302]:
```suggestion
    const int64 actual_time = instructions_elapsed_time.size();
    if (logical_time >= actual_time) {
```

File: tensorflow/compiler/xla/service/memory_space_assignment.cc [hunk 3]
```diff
@@ -2856,8 +2857,8 @@ Status MemorySpaceAssignment::FixSchedule() {
 
     VLOG(4) << "Scheduling: " << computation->ToString();
 
-    for (int64 instruction_index = 0;
-         instruction_index < flattened_instructions_.size();
+    for (int64 instruction_index = 0, end = flattened_instructions_.size();
+         instruction_index < end;
          ++instruction_index) {
```
Comment (mihaimaruseac) [inline tensorflow/compiler/xla/service/memory_space_assignment.cc:2862]: These two lines can now be merged

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

**Comment by tg-at-google** [comment]
> `tensorflow/compiler/xla` is also covered y #41752.

Yes, they are essentially 2 batches.

File: tensorflow/compiler/xla/service/llvm_ir/llvm_loop.cc (hunk 1 above)
Comment (tg-at-google) [reply tensorflow/compiler/xla/service/llvm_ir/llvm_loop.cc:281]: update appled.

File: tensorflow/compiler/xla/service/memory_space_assignment.cc (hunk 3 above)
Comment (tg-at-google) [reply tensorflow/compiler/xla/service/memory_space_assignment.cc:2862]: update appled.

**Comment by tg-at-google** [comment]
updates applied.

File: tensorflow/compiler/xla/service/llvm_ir/llvm_loop.cc [hunk 4]
```diff
@@ -276,7 +276,7 @@ std::vector<llvm::Value*> ForLoopNest::EmitOperandArrayLoopNest(
       AddLoopsForShapeOnDimensions(shape, dimensions, name_suffix);
   // Verify every dimension except the 'dimension_to_skip' dimension was set in
   // the index.
-  for (size_t dimension = 0; dimension < multi_index.size(); ++dimension) {
+  for (int64 dimension = 0; dimension < multi_index.size(); ++dimension) {
```
Comment (mihaimaruseac) [inline tensorflow/compiler/xla/service/llvm_ir/llvm_loop.cc:279]: I don't think we need to change anything here. Both the loop counter and the limit should be unsigned

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

**Comment by tg-at-google** [comment]
updates applied, 2.

**Review by mihaimaruseac — APPROVED** [review]

**Comment by mihaimaruseac** [comment]
This seems to cause build failures. Can you investigate please?

**Comment by tg-at-google** [comment]
> This seems to cause build failures. Can you investigate please?

++, will investigate.

**Pull request closed without merging**